doc/adrs/053_add_support_for_form_descriptions_in_the_view_converter.adoc

```diff
@@ -0,0 +1,43 @@
+= ADR-053 - Add support for form descriptions in the view converter
+
+== Context
+
+The RepresentationDescriptions in the View DSL needs to be converted in the supported type `org.eclipse.sirius.components.representations.IRepresentationDescription`.
+For now, only `DiagramDescription` are converted by the `ViewConverter`.
+
+== Decision
+
+We need to create and add a `FormDescriptionConverter` to the `ViewConverter`.
+The `ViewConverter.convert()` method statically converts DiagramDescription:
+
+```            
+result = view.getDescriptions().stream()
+    .filter(org.eclipse.sirius.components.view.DiagramDescription.class::isInstance)
+    .map(org.eclipse.sirius.components.view.DiagramDescription.class::cast)
+    .map(viewDiagramDescription -> this.convertDiagramDescription(viewDiagramDescription, interpreter))
+    .collect(Collectors.toList());
+                         
+```
+
+We need to modify this logic to dispatch to the correct converter according to the `IRepresentationDescription` concrete type.
+We will create a common interface `IRepresentationDescriptionConverter` for `FormDescriptionConverter` and `DiagramDescriptionConverter`. The `ViewConverter` will no longer reference a specific converter implementation.
+
+The `FormDescriptionConverter` will create a new `org.eclipse.sirius.components.forms.description.FormDescription`. The DSL does not yet handle Pages and Groups. The converter will create a default page and group.
+
+As a first step, we will only convert `TextfieldDescription`. 
+
+Since the View DSL does not yet allow defining behavior on the Textfield Widget, the following fields from `TextfieldDescription` will be provided with default values:
+
+-`textfieldDescription.newValueHandler`
```

Comment:
`TextfieldDescription` does not seem to be tied to a specific attribute, but use an arbitrary `valueExpression`.
Can you clarify what would be the default behavior of the `newValueHandler`? I.e. how do you determine which attribute of which semantic element to `eSet` the new value?

Reply:
OK, so I just read the last sentence below:

> without editing capabilities in a first step

So my remark does not apply to this version, but the fact that the current scope does not include edition could be mentioned earlier.

Reply:
Good remark, I will clarify the "default" behavior for each one of them.

Reply:
> `TextfieldDescription` does not seem to be tied to a specific attribute, but use an arbitrary `valueExpression`.
> Can you clarify what would be the default behavior of the `newValueHandler`? I.e. how do you determine which attribute of which semantic element to `eSet` the new value?

That's the same behavior as in EEF / Sirius Desktop. I'm not sure that we want to change that to tightly coupled an attribute with a widget.

We could decide that we don't want to have the same behavior as in Sirius and couple widgets and attributes together (which would probably make simple use cases easier and make advanced use cases more complex) but that's a major change from the way things have been working so far. The label of a node or and edge in the view is not linked to an attribute either for example.

Reply:
From what I understand:
* The value of the textfield is computed according to the value expression provided by the specifier.
*  The behavior when changing this value will be specified in an another PR (by updating this ADR ?) but the specifier will also describe (with view::Operation probably), what is done with the new value entered by the end-user.

Reply:
@sbegaudeau yes I understand, my remark was that on first reading "we will provide a default value for the newValueHandler" begged for the question: what kind of default behavior can you determine given that the text value is *not* bound to a specific attribute but computed? The ADR mentions that the first iteration will be read-only, but this is only mentioned later. I'm not asking for changing the behavior, just to clarify the text, for example just making it clear that the default behavior for newValueHandler does nothing.

Reply:
I have not read the ADR yet, so I did not see that part in the ADR:  "we will provide a default value for the newValueHandler" :)

I want to be also clear, I do not know for a fact that we want the existing Sirius Desktop / EEF behavior, I need to clarify that first. We could very well end up with widgets tightly connected to features and a Form Description view DSL made with beginners in mind.

I'm pretty sure that we did not write anywhere what we want on that subject (on purpose to have this conversation, if I remember correctly). So in the end, I just know that we don't know... or at least that I don't know 😄